lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_helper.rb

```diff
@@ -1,11 +1,57 @@
 require 'fastlane_core/ui/ui'
-require 'cfpropertylist'
+require_relative '../actions/firebase_app_distribution_login'
 
 module Fastlane
   UI = FastlaneCore::UI unless Fastlane.const_defined?("UI")
 
   module Helper
     module FirebaseAppDistributionHelper
```

Comment:
I'm not sure you're gaining much by creating helpers to get the endpoint paths, connection and auth_token. The `FirebaseAppDistributionAction` still needs to how to use all those things e.g. call `connection.get(upload_status_path(app_id, app_token))` to get the upload status. I think it would be more useful to create a separate `FirebaseAppDistributionApiClient` class that will handle all the network calls. Then the the `run` method can just call `upload`, and the `FirebaseAppDistributionApiClient`, which contains all the necessary data, will make those calls.

Reply:
Agree with Tunde's suggestion to create an API client class. So far the helper has been essentially a place to put some of the action code so that the action is fewer lines, but because there's no real difference between the two it's hard for the reader to understand what goes where, or why they are separate.

Instead, abstracting all of the network-related behavior into one class would be a clear separation of concerns: the client's job is to talk to the API, and that's what the action uses it for.

I do like the new URL helper methods though 👍 . I think it's nice to have them all next to each other.